XiEditor/EditViewController.swift

```diff
@@ -516,10 +516,19 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
             mouseDragged(with: event)
         }
     }
-    
-    // used mostly for paste
+
+    // NOTE: this was previously used for paste, and could possibly be removed, but it
+    // may be used for IME?
     override func insertText(_ insertString: Any) {
-        document.sendRpcAsync("insert", params: insertedStringToJson(insertString as! NSString))
+        let text: String
+        if let s = insertString as? NSString {
```

Comment:
You can write this check as `insertString is NSString` and it would look a bit more verbose.